pkg/run/run.go

```diff
@@ -175,77 +176,128 @@ func (r *RunOptions) PreRunCheck(kclient kubernetes.Interface) error {
 
 	// sonobuoy namespace exists so return error
 	if p.Name != "" {
-		return errors.New("sonobuoy namespace already exists")
+		return errors.New(fmt.Sprintf("%s namespace already exists", pkg.CertificationNamespace))
+	}
+
+	namespace := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   pkg.CertificationNamespace,
+			Labels: pkg.SonobuoyDefaultLabels,
+		},
 	}
 
 	if r.dedicated {
+
 		log.Info("Ensuring proper node label for dedicated mode")
 		nodes, err := coreClient.Nodes().List(context.TODO(), metav1.ListOptions{
-			LabelSelector: "node-role.kubernetes.io/tests=",
+			LabelSelector: pkg.DedicatedNodeRoleLabelSelector,
 		})
 		if err != nil {
 			return err
 		}
 		if nodes.Items != nil && len(nodes.Items) == 0 {
 			return errors.New("No nodes with role required for dedicated mode (node-role.kubernetes.io/tests)")
 		}
+
+		tolerations, err := json.Marshal([]v1.Toleration{{
+			Key:      pkg.DedicatedNodeRoleLabel,
+			Operator: v1.TolerationOpExists,
+			Value:    "",
+			Effect:   v1.TaintEffectNoSchedule,
+		}})
+		if err != nil {
+			return err
+		}
+
+		namespace.Annotations = map[string]string{
+			"scheduler.alpha.kubernetes.io/defaultTolerations": string(tolerations),
+			"openshift.io/node-selector":                       pkg.DedicatedNodeRoleLabelSelector,
+		}
+	}
+
+	_, err = kclient.CoreV1().Namespaces().Create(context.TODO(), namespace, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error creating Namespace")
+	}
+
+	// Create Sonobuoy ServiceAccount
+	// https://github.com/vmware-tanzu/sonobuoy/blob/main/pkg/client/gen.go#L611-L616
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pkg.SonobuoyServiceAccountName,
+			Namespace: pkg.CertificationNamespace,
+			Labels:    pkg.SonobuoyDefaultLabels,
+		},
+	}
+	sa.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ServiceAccount"})
```

Comment:
as the `schema.GroupVersionKind` has been expanded in the other lines ([263](https://github.com/redhat-openshift-ecosystem/provider-certification-tool/pull/34/files#diff-bb95634405f8d047841128dbd79e9058b927a35a23011aa7096d899c21764896R263), [294](https://github.com/redhat-openshift-ecosystem/provider-certification-tool/pull/34/files#diff-bb95634405f8d047841128dbd79e9058b927a35a23011aa7096d899c21764896R294)) it would be fine to expand here too.